[
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/524423874",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-524423874",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 524423874,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUyNDQyMzg3NA==",
        "user": {
            "login": "faxu",
            "id": 20780999,
            "node_id": "MDQ6VXNlcjIwNzgwOTk5",
            "avatar_url": "https://avatars.githubusercontent.com/u/20780999?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/faxu",
            "html_url": "https://github.com/faxu",
            "followers_url": "https://api.github.com/users/faxu/followers",
            "following_url": "https://api.github.com/users/faxu/following{/other_user}",
            "gists_url": "https://api.github.com/users/faxu/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/faxu/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/faxu/subscriptions",
            "organizations_url": "https://api.github.com/users/faxu/orgs",
            "repos_url": "https://api.github.com/users/faxu/repos",
            "events_url": "https://api.github.com/users/faxu/events{/privacy}",
            "received_events_url": "https://api.github.com/users/faxu/received_events",
            "type": "User",
            "site_admin": true
        },
        "created_at": "2019-08-23T18:50:24Z",
        "updated_at": "2019-08-23T18:50:24Z",
        "author_association": "MEMBER",
        "body": "@ebarsoumMS : can you see if this is an issue with cntk export? ",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/524423874/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/525216803",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-525216803",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 525216803,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUyNTIxNjgwMw==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-08-27T09:18:28Z",
        "updated_at": "2019-08-27T09:18:28Z",
        "author_association": "NONE",
        "body": "The same file actually has issues reading in CNTK, although a bugfix to the CNTK reader sorts that. Worth linking the issue here:\r\n\r\nhttps://github.com/microsoft/CNTK/issues/3746",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/525216803/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/529555156",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-529555156",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 529555156,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUyOTU1NTE1Ng==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-09-09T16:16:37Z",
        "updated_at": "2019-09-09T16:16:37Z",
        "author_association": "NONE",
        "body": "@ebarsoumMS any progress on this? Is it likely a CNTK issue or a c++ onnxruntime issue? I had a quick look at the CNTK code and it seems to use onnx runtime (possibly not the latest version??) to do the export. ",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/529555156/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532121754",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-532121754",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 532121754,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUzMjEyMTc1NA==",
        "user": {
            "login": "huangyiping-ai",
            "id": 55434356,
            "node_id": "MDQ6VXNlcjU1NDM0MzU2",
            "avatar_url": "https://avatars.githubusercontent.com/u/55434356?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/huangyiping-ai",
            "html_url": "https://github.com/huangyiping-ai",
            "followers_url": "https://api.github.com/users/huangyiping-ai/followers",
            "following_url": "https://api.github.com/users/huangyiping-ai/following{/other_user}",
            "gists_url": "https://api.github.com/users/huangyiping-ai/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/huangyiping-ai/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/huangyiping-ai/subscriptions",
            "organizations_url": "https://api.github.com/users/huangyiping-ai/orgs",
            "repos_url": "https://api.github.com/users/huangyiping-ai/repos",
            "events_url": "https://api.github.com/users/huangyiping-ai/events{/privacy}",
            "received_events_url": "https://api.github.com/users/huangyiping-ai/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-09-17T08:38:11Z",
        "updated_at": "2019-09-17T08:38:11Z",
        "author_association": "NONE",
        "body": "I have the same problem. Have you solved it?",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532121754/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532136791",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-532136791",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 532136791,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUzMjEzNjc5MQ==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-09-17T09:18:00Z",
        "updated_at": "2019-09-17T09:18:00Z",
        "author_association": "NONE",
        "body": "@huangyiping-ai no, not yet. I'm still awaiting the guys from Microsoft to comment whether it's likely a cntk or onnxruntime issue. Would really like a solution!",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532136791/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532149072",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-532149072",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 532149072,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUzMjE0OTA3Mg==",
        "user": {
            "login": "huangyiping-ai",
            "id": 55434356,
            "node_id": "MDQ6VXNlcjU1NDM0MzU2",
            "avatar_url": "https://avatars.githubusercontent.com/u/55434356?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/huangyiping-ai",
            "html_url": "https://github.com/huangyiping-ai",
            "followers_url": "https://api.github.com/users/huangyiping-ai/followers",
            "following_url": "https://api.github.com/users/huangyiping-ai/following{/other_user}",
            "gists_url": "https://api.github.com/users/huangyiping-ai/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/huangyiping-ai/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/huangyiping-ai/subscriptions",
            "organizations_url": "https://api.github.com/users/huangyiping-ai/orgs",
            "repos_url": "https://api.github.com/users/huangyiping-ai/repos",
            "events_url": "https://api.github.com/users/huangyiping-ai/events{/privacy}",
            "received_events_url": "https://api.github.com/users/huangyiping-ai/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-09-17T09:51:27Z",
        "updated_at": "2019-09-17T09:51:27Z",
        "author_association": "NONE",
        "body": "@dmagee \r\nI converted the pytorch model into onnx model.\r\nWhen I run\r\nImport onnx\r\nModel_onnx_path= \"LPRNet_model.onnx\"\r\nModel = onnx. load (model_onnx_path)\r\nOnnx. checker. check_model (model)\r\nWhen onnx version = 1.1.? The shape error of onnx. checker. check_model\r\nWhen onnx version = 1.3.0, onnx. checker. check_model passes\r\nSo, I guess on onnxruntime, is it also a version error? For reference only, the guess may be wrong.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532149072/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532194742",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-532194742",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 532194742,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUzMjE5NDc0Mg==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-09-17T12:17:20Z",
        "updated_at": "2019-09-17T12:17:20Z",
        "author_association": "NONE",
        "body": "@huangyping-ai that looks like you are using the python version of onnxruntime. My model works with the python version, just not with the c++ version. I think you probably have a slightly different issue? I may be wrong.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/532194742/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/537612631",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-537612631",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 537612631,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUzNzYxMjYzMQ==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-10-02T18:06:07Z",
        "updated_at": "2019-10-02T18:06:07Z",
        "author_association": "NONE",
        "body": "Update: I'm not sure if anyone at Microsoft is looking into this (@faxu, @ebarsoumMS ?), but I downloaded the  onnxruntime source. The C API seems to be in a state of flux (my example and one of the provided examples doesn't compile due to various missing functions), but I got a C++ example working and it gives the same error as the binary release with c API:\r\n\r\n Load model from TISSUETYPE_8X_DSNET_12D.cntk2.7.onnx failed:[ShapeInferenceError] Attribute strides has incorrect size\r\n\r\nI tracked this down to old.cc ~ line 60:\r\n\r\n```\r\n  std::vector<int64_t> strides;\r\n  if (getRepeatedAttribute(ctx, \"strides\", strides)) {\r\n    if (strides.size() != n_input_dims) {\r\n      fail_shape_inference(\"Attribute strides has incorrect size\");\r\n    }\r\n  } else {\r\n    strides.assign(n_input_dims, 1);\r\n  }\r\n```\r\n\r\nstrides.size() seems to be 2, whereas n_input_dims is 3. The latter is correct I think. The input is an RGB image so 3 x 1024 x 1024. All functions in the network deal with 3D data (w*h*rgb or w*h*nfmaps).\r\n\r\nStill not sure how to fix it though. CNTK uses onnxruntime to write the file.\r\n\r\nD.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/537612631/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/537944960",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-537944960",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 537944960,
        "node_id": "MDEyOklzc3VlQ29tbWVudDUzNzk0NDk2MA==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-10-03T13:26:18Z",
        "updated_at": "2019-10-03T13:40:03Z",
        "author_association": "NONE",
        "body": "Further update: This seems to be an issue with pooling node (convPoolShapeInference1(), old.cc). This network does 3x3 max pooling on 3D data (W x H x RGB or later  W x H x nFmaps). To achieve this in CNTK you provide a 2D shape description (CNTK::NDShape(2, 3) in C++). Obviously this has an implicit 3rd dimension (the kernel is effectively 3x3x1). The same issue applies to strides ([2,2] and pad as well i.e. they are specified as 2D, but are implicitly 3D. It seems they are output to the onnx file in this implicit form. onnxruntime doesn't seem to handle this when reading the file. I tried adding a few lines to the parser (in old.cc) in the appropriate places between load and checks:\r\n\r\n    while(strides.size() < n_input_dims) {\r\n        strides.push_back(1) ;\r\n    }\r\n...\r\n      while (kernel_shape.size() < n_input_dims) {\r\n          kernel_shape.push_back(1);\r\n      }\r\n...\r\n      while (pads.size() < n_input_dims*2) {\r\n          pads.push_back(0);\r\n      }\r\n\r\nHowever, now I'm getting the error:\r\n\r\nfailed:Node:Pooling86 Output:Pooling86_Output_0 [ShapeInferenceError] Can't merge shape info. Both source and target dimension have values but they differ. Source=15 Target=32 Dimension=2\r\n\r\nThis pool should go from 1x32x1024x512 -> 1 x 32x511x255 (i.e. no padding in any dimension, and no pooling in the fmaps dimension), however my change above to the kernel shape seems to make it pool in that dimension as well causing the error (at least that's what I think is happening).\r\n\r\nI still have no idea if this is an onnx issue or a cntk one (I guess that depends on what's in the onnx standard about implicit dimensions - is there even a standard??). However, netron and the python version of onnx seem to handle this fine.  \r\n\r\n\r\n\r\n\r\n  ",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/537944960/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/552637271",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-552637271",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 552637271,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1MjYzNzI3MQ==",
        "user": {
            "login": "faxu",
            "id": 20780999,
            "node_id": "MDQ6VXNlcjIwNzgwOTk5",
            "avatar_url": "https://avatars.githubusercontent.com/u/20780999?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/faxu",
            "html_url": "https://github.com/faxu",
            "followers_url": "https://api.github.com/users/faxu/followers",
            "following_url": "https://api.github.com/users/faxu/following{/other_user}",
            "gists_url": "https://api.github.com/users/faxu/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/faxu/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/faxu/subscriptions",
            "organizations_url": "https://api.github.com/users/faxu/orgs",
            "repos_url": "https://api.github.com/users/faxu/repos",
            "events_url": "https://api.github.com/users/faxu/events{/privacy}",
            "received_events_url": "https://api.github.com/users/faxu/received_events",
            "type": "User",
            "site_admin": true
        },
        "created_at": "2019-11-11T22:03:29Z",
        "updated_at": "2019-11-11T22:03:29Z",
        "author_association": "MEMBER",
        "body": "@dmagee @huangyiping-ai are you still impacted by this issue with the latest version of ONNX Runtime? If so and you still require assistance, please let us know and we can investigate further. ",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/552637271/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/554605299",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-554605299",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 554605299,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NDYwNTI5OQ==",
        "user": {
            "login": "YanYas",
            "id": 9497052,
            "node_id": "MDQ6VXNlcjk0OTcwNTI=",
            "avatar_url": "https://avatars.githubusercontent.com/u/9497052?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/YanYas",
            "html_url": "https://github.com/YanYas",
            "followers_url": "https://api.github.com/users/YanYas/followers",
            "following_url": "https://api.github.com/users/YanYas/following{/other_user}",
            "gists_url": "https://api.github.com/users/YanYas/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/YanYas/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/YanYas/subscriptions",
            "organizations_url": "https://api.github.com/users/YanYas/orgs",
            "repos_url": "https://api.github.com/users/YanYas/repos",
            "events_url": "https://api.github.com/users/YanYas/events{/privacy}",
            "received_events_url": "https://api.github.com/users/YanYas/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-16T05:06:44Z",
        "updated_at": "2019-11-16T05:06:44Z",
        "author_association": "NONE",
        "body": "Hi @faxu , I've come across this problem too. I'll test with the latest version of Onnx and report back.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/554605299/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/554606428",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-554606428",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 554606428,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NDYwNjQyOA==",
        "user": {
            "login": "YanYas",
            "id": 9497052,
            "node_id": "MDQ6VXNlcjk0OTcwNTI=",
            "avatar_url": "https://avatars.githubusercontent.com/u/9497052?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/YanYas",
            "html_url": "https://github.com/YanYas",
            "followers_url": "https://api.github.com/users/YanYas/followers",
            "following_url": "https://api.github.com/users/YanYas/following{/other_user}",
            "gists_url": "https://api.github.com/users/YanYas/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/YanYas/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/YanYas/subscriptions",
            "organizations_url": "https://api.github.com/users/YanYas/orgs",
            "repos_url": "https://api.github.com/users/YanYas/repos",
            "events_url": "https://api.github.com/users/YanYas/events{/privacy}",
            "received_events_url": "https://api.github.com/users/YanYas/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-16T05:29:14Z",
        "updated_at": "2019-11-16T05:29:14Z",
        "author_association": "NONE",
        "body": "\r\n```\r\n[ErrorCode:Fail] Load model from C:\\Users\\Igor\\Documents\\VLML\\PythonResearchProjects\\PythonExportCNTKToOnnx\\VAE_Model_4.onnx failed:[ShapeInferenceError] Attribute dilations has incorrect size\r\n   at Microsoft.ML.OnnxRuntime.NativeApiStatus.VerifySuccess(IntPtr nativeStatus)\r\n   at Microsoft.ML.OnnxRuntime.InferenceSession.Init(String modelPath, SessionOptions options)\r\n   at Microsoft.ML.OnnxRuntime.InferenceSession..ctor(String modelPath, SessionOptions options)\r\n   at VLML.RunnerCore.UpdateDEBUG(PatchTracer tracer, Path ModelFile, IEnumerable`1 Input_Data...\r\n```\r\nSame error for two different models (one a [VAE ](https://drive.google.com/file/d/1judW2Xw2JlKdJRt6JSWMv57PPRQ6_wJt/view?usp=sharing) the other a ResNet Transfer Learning example which I'll link to when its finished uploading.\r\n\r\nThanks for looking into this",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/554606428/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/554614258",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-554614258",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 554614258,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU1NDYxNDI1OA==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-11-16T07:49:47Z",
        "updated_at": "2019-11-16T07:49:47Z",
        "author_association": "NONE",
        "body": "I thought I'd commented last week, but yes it is still an issue in the latest binary release.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/554614258/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/566769995",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-566769995",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 566769995,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU2Njc2OTk5NQ==",
        "user": {
            "login": "ke1337",
            "id": 22626095,
            "node_id": "MDQ6VXNlcjIyNjI2MDk1",
            "avatar_url": "https://avatars.githubusercontent.com/u/22626095?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/ke1337",
            "html_url": "https://github.com/ke1337",
            "followers_url": "https://api.github.com/users/ke1337/followers",
            "following_url": "https://api.github.com/users/ke1337/following{/other_user}",
            "gists_url": "https://api.github.com/users/ke1337/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/ke1337/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/ke1337/subscriptions",
            "organizations_url": "https://api.github.com/users/ke1337/orgs",
            "repos_url": "https://api.github.com/users/ke1337/repos",
            "events_url": "https://api.github.com/users/ke1337/events{/privacy}",
            "received_events_url": "https://api.github.com/users/ke1337/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2019-12-17T22:05:12Z",
        "updated_at": "2019-12-17T22:20:40Z",
        "author_association": "CONTRIBUTOR",
        "body": "From ONNX [spec](https://github.com/onnx/onnx/blob/master/docs/Operators.md#MaxPool) and [implementation](https://github.com/onnx/onnx/blob/master/onnx/defs/nn/defs.cc#L138-L140), it seems to only support pooling2D for 4D input, and pooling3D for 5D input. The model exported from CNTK has a 5D input for pooling ops, but the kernel shape is 2D. In the same model, there are conv ops having reshape before/after to make sure input is 4D, and IMO Pooling ops should do the same in CNTK exporter.\r\n\r\nFor Netron and python onnx module, shape inference is either not invoked or silently ignored. Before the bug is fixed, a workaround is to edit the MaxPool/AveragePool nodes in the model, with Squeeze/Unsqueeze (assuming batch==1) before and after to make sure the input to pooling ops are 4D:\r\n```\r\nimport onnx\r\nfrom onnx import helper, shape_inference\r\nmp = onnx.load('/path/to/model')\r\nmp_fix = onnx.ModelProto()\r\nmp_fix.CopyFrom(mp)\r\nmp_fix.graph.ClearField('node')\r\nmp_fix.graph.ClearField('value_info')\r\nfor n in mp.graph.node:\r\n  if n.op_type in ['MaxPool', 'AveragePool']:\r\n    mp_fix.graph.node.add().CopyFrom(helper.make_node('Squeeze', n.input, [n.input[0] + '_squeezed'], n.name + '_squeeze_before', axes=[1]))\r\n    pool_node = mp_fix.graph.node.add()\r\n    pool_node.CopyFrom(n)\r\n    pool_node.input[0] += '_squeezed'\r\n    pool_node.output[0] += '_before_unsqueeze'\r\n    mp_fix.graph.node.add().CopyFrom(helper.make_node('Unsqueeze', pool_node.output, n.output, n.name + '_unsqueeze_after', axes=[1]))\r\n  else:\r\n    mp_fix.graph.node.add().CopyFrom(n)\r\nmp_fix = shape_inference.infer_shapes(mp_fix)\r\nonnx.save(mp_fix, '/path/to/fixed_model')\r\n```\r\n\r\nNow the fixed model should have no issue in loading in ONNX runtime.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/566769995/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/592255906",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-592255906",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 592255906,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU5MjI1NTkwNg==",
        "user": {
            "login": "faxu",
            "id": 20780999,
            "node_id": "MDQ6VXNlcjIwNzgwOTk5",
            "avatar_url": "https://avatars.githubusercontent.com/u/20780999?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/faxu",
            "html_url": "https://github.com/faxu",
            "followers_url": "https://api.github.com/users/faxu/followers",
            "following_url": "https://api.github.com/users/faxu/following{/other_user}",
            "gists_url": "https://api.github.com/users/faxu/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/faxu/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/faxu/subscriptions",
            "organizations_url": "https://api.github.com/users/faxu/orgs",
            "repos_url": "https://api.github.com/users/faxu/repos",
            "events_url": "https://api.github.com/users/faxu/events{/privacy}",
            "received_events_url": "https://api.github.com/users/faxu/received_events",
            "type": "User",
            "site_admin": true
        },
        "created_at": "2020-02-28T00:58:39Z",
        "updated_at": "2020-02-28T00:58:39Z",
        "author_association": "MEMBER",
        "body": "@dmagee does the fixed model work for you? It seems the root cause of the issue will take some time to investigate in CNTK and is lower priority at this time. Let us know if you're blocked on this with some urgency. ",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/592255906/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/592559838",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-592559838",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 592559838,
        "node_id": "MDEyOklzc3VlQ29tbWVudDU5MjU1OTgzOA==",
        "user": {
            "login": "dmagee",
            "id": 6396177,
            "node_id": "MDQ6VXNlcjYzOTYxNzc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/6396177?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/dmagee",
            "html_url": "https://github.com/dmagee",
            "followers_url": "https://api.github.com/users/dmagee/followers",
            "following_url": "https://api.github.com/users/dmagee/following{/other_user}",
            "gists_url": "https://api.github.com/users/dmagee/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/dmagee/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/dmagee/subscriptions",
            "organizations_url": "https://api.github.com/users/dmagee/orgs",
            "repos_url": "https://api.github.com/users/dmagee/repos",
            "events_url": "https://api.github.com/users/dmagee/events{/privacy}",
            "received_events_url": "https://api.github.com/users/dmagee/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2020-02-28T15:20:44Z",
        "updated_at": "2020-02-28T15:20:44Z",
        "author_association": "NONE",
        "body": "I'm afraid we abandoned trying to get onnxruntime to work. We're working in a commercial environment and the fact that you couldn't sort this properly (even given the fact that we practically gave you the solution above) in a sensible timeframe meant we went looking for other solutions, Our networks work with TensorRT now (as of v7) and we're going forward with that, which has the added advantage of native TensorFlow support when we eventually move away from CNTK. \r\n\r\nIf you want to fix onnxruntime it for others I suggest looking at my comments above. I think the issue with the \"Can't merge shape info.\" was that I added the implicit dimensions at the wrong end of the array. I suspect it's pretty easy to get onnxruntime to support this, and given both CNTK and pyTorch have generated such models possibly a good idea?",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/592559838/reactions",
            "total_count": 2,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 2,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/653347587",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-653347587",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 653347587,
        "node_id": "MDEyOklzc3VlQ29tbWVudDY1MzM0NzU4Nw==",
        "user": {
            "login": "stale[bot]",
            "id": 26384082,
            "node_id": "MDM6Qm90MjYzODQwODI=",
            "avatar_url": "https://avatars.githubusercontent.com/in/1724?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/stale%5Bbot%5D",
            "html_url": "https://github.com/apps/stale",
            "followers_url": "https://api.github.com/users/stale%5Bbot%5D/followers",
            "following_url": "https://api.github.com/users/stale%5Bbot%5D/following{/other_user}",
            "gists_url": "https://api.github.com/users/stale%5Bbot%5D/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/stale%5Bbot%5D/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/stale%5Bbot%5D/subscriptions",
            "organizations_url": "https://api.github.com/users/stale%5Bbot%5D/orgs",
            "repos_url": "https://api.github.com/users/stale%5Bbot%5D/repos",
            "events_url": "https://api.github.com/users/stale%5Bbot%5D/events{/privacy}",
            "received_events_url": "https://api.github.com/users/stale%5Bbot%5D/received_events",
            "type": "Bot",
            "site_admin": false
        },
        "created_at": "2020-07-03T04:58:09Z",
        "updated_at": "2020-07-03T04:58:09Z",
        "author_association": "NONE",
        "body": "This issue has been automatically marked as stale due to inactivity and will be closed in 7 days if no further activity occurs. If further support is needed, please provide an update and/or more details.\n",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/653347587/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/656971866",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/1685#issuecomment-656971866",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/1685",
        "id": 656971866,
        "node_id": "MDEyOklzc3VlQ29tbWVudDY1Njk3MTg2Ng==",
        "user": {
            "login": "stale[bot]",
            "id": 26384082,
            "node_id": "MDM6Qm90MjYzODQwODI=",
            "avatar_url": "https://avatars.githubusercontent.com/in/1724?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/stale%5Bbot%5D",
            "html_url": "https://github.com/apps/stale",
            "followers_url": "https://api.github.com/users/stale%5Bbot%5D/followers",
            "following_url": "https://api.github.com/users/stale%5Bbot%5D/following{/other_user}",
            "gists_url": "https://api.github.com/users/stale%5Bbot%5D/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/stale%5Bbot%5D/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/stale%5Bbot%5D/subscriptions",
            "organizations_url": "https://api.github.com/users/stale%5Bbot%5D/orgs",
            "repos_url": "https://api.github.com/users/stale%5Bbot%5D/repos",
            "events_url": "https://api.github.com/users/stale%5Bbot%5D/events{/privacy}",
            "received_events_url": "https://api.github.com/users/stale%5Bbot%5D/received_events",
            "type": "Bot",
            "site_admin": false
        },
        "created_at": "2020-07-11T02:39:30Z",
        "updated_at": "2020-07-11T02:39:30Z",
        "author_association": "NONE",
        "body": "This issue has been automatically closed due to inactivity. Please reactivate if further support is needed.\n",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/656971866/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    }
]